code/src/gfx.c

```diff
@@ -16,24 +16,99 @@ static u32 closingButton = 0;
 static u8 currentSphere = 0;
 static s16 spoilerScroll = 0;
 static s16 allItemsScroll = 0;
+static s16 groupItemsScroll = 0;
+static s8 currentGroup = 0;
 static s32 curMenuIdx = 0;
 static float itemPercent = 0;
 
+static char *spoilerCollectionGroupNames[] = {
+    "",
+    "Kokiri Forest",
+    "Lost Woods",
+    "Deku Tree",
+    "Forest Temple",
+    "Kakariko Village",
+    "Bottom of the Well",
+    "Shadow Temple",
+    "Death Mountain",
+    "Goron City",
+    "Dodongo's Cavern",
+    "Fire Temple",
+    "Zora's River",
+    "Zora's Domain",
+    "Jabu Jabu's Belly",
+    "Ice Cavern",
+    "Hyrule Field",
+    "Lon Lon Ranch",
+    "Lake Hylia",
+    "Water Temple",
+    "Gerudo Valley",
+    "Gerudo Training Grounds",
+    "Spirit Temple",
+    "Hyrule Castle",
+    "Ganon's Castle",
+};
+
 #define UP_ARROW_CHR 24
 #define DOWN_ARROW_CHR 25
 #define LEFT_ARROW_CHR 27
 #define RIGHT_ARROW_CHR 26
 #define UP_SOLID_ARROW_CHR 30
 #define DOWN_SOLID_ARROW_CHR 31
 
-#define MAX_ITEM_LINES 6
+#define MAX_ITEM_LINES 8
+#define SCROLL_BAR_THICKNESS 2
+#define SCROLL_BAR_MIN_THUMB_SIZE 4
 #define COLOR_WARN RGB8(0xD1, 0xDF, 0x3C)
+#define COLOR_SCROLL_BAR_BG RGB8(0x58, 0x58, 0x58)
+
+static void Gfx_DrawScrollBar(u16 barX, u16 barY, u16 barSize, u16 currentScroll, u16 maxScroll, u16 pageSize) {
+    Draw_DrawRect(barX, barY, SCROLL_BAR_THICKNESS, barSize, COLOR_SCROLL_BAR_BG);
+
+    float thumbSizePercent = pageSize / (float)maxScroll;
+    if (thumbSizePercent > 1.0f) {
+        thumbSizePercent = 1.0f;
+    }
+    u16 thumbSize = (u16)(thumbSizePercent * barSize);
+    if (thumbSize < SCROLL_BAR_MIN_THUMB_SIZE) {
+        thumbSize = SCROLL_BAR_MIN_THUMB_SIZE;
+    }
+    float barThumbPosPercent = (float)currentScroll / (float)(maxScroll - pageSize);
+    u16 barThumbPosY = (u16)(barThumbPosPercent * (barSize - thumbSize));
+    Draw_DrawRect(barX, barY + barThumbPosY, SCROLL_BAR_THICKNESS, thumbSize, COLOR_WHITE);
+}
+
+static void Gfx_GroupsNextGroup() {
+    groupItemsScroll = 0;
+    s8 prevGroup = currentGroup;
+    do {
+        ++currentGroup;
+        if (currentGroup >= SPOILER_COLLECTION_GROUP_COUNT) { currentGroup = 1; }
+    } while (gSpoilerData.GroupItemCounts[currentGroup] == 0 && currentGroup != prevGroup);

```

Comment:
```suggestion
    do {
        ++currentGroup;
        if (currentGroup >= SPOILER_COLLECTION_GROUP_COUNT) { currentGroup = 1; }
    } while (gSpoilerData.GroupItemCounts[currentGroup] == 0);

```
Unused variables? Removing them still works. Same for `Gfx_GroupsPrevGroup`.